Source/WebCore/dom/IdleDeadline.cpp

```diff
@@ -38,9 +38,9 @@ DOMHighResTimeStamp IdleDeadline::timeRemaining(Document& document) const
     RefPtr window { document.domWindow() };
     if (!window || m_didTimeout == DidTimeout::Yes)
         return 0;
-    auto deadline = document.windowEventLoop().computeIdleDeadline();
-    auto remainingTime = window->performance().relativeTimeFromTimeOriginInReducedResolution(deadline) - window->performance().now();
-    return remainingTime < 0 ? 0 : remainingTime;
+    auto duration = document.windowEventLoop().computeIdleDeadline() - MonotonicTime::now();
```

Comment:
While this `MonotonicTime::now()` is not new since it came from `Performance::nowInReducedResolutionSeconds(…)`, I wonder if we could use `ApproximateTime` instead?

Reply:
We can't use ApproximateTime for this since this value will be exposed to JS via IdleDeadline.timeRemaining.